### PR TITLE
RIA-5721 Enable integration with F&P

### PR DESCRIPTION
**Pull request by Marcellofabbri**
### JIRA link (if applicable) ###
[RIA-5721](https://tools.hmcts.net/jira/browse/RIA-5721)

**The PR doesn't change the current payment functionality, which is supposed to be altered in a following ticket. This is just to enable WaysToPay**

**A number of vulnerability suppressions were added**

### Change description ###
- `ServiceRequestApi` was created. It sends an API call to the common component to create a service-request
- `ServiceRequestService` was created, and it uses a `ServiceRequestApi` to send an API call to create a service-request

- isWaysToPay boolean type has been added to `PaymentAppealPreparer` (event=START_APPEAL, stage=ABOUT_TO_SUBMIT, isLegalRep=TRUE)
- `PaymentAppealPreparer` has been changed so that when isWaysToPay is true it uses `ServiceRequestService` to create a service-request

- `ServiceRequestUpdateController` was created with an endpoint that some common component is supposed to call when the payment goes through. Its role is to receive confirmation of the payment and use the details of that confirmation to update the asylum case

- all the necessary dtos were added

- lombok.config was added

- to attempt to make it work in other environments, some environment variables probably need to be stored in the key vault, namely: 
    - PAYMENT_API_URL (the common component, which is locally is payments-api on localhost:8083)
    - PAY_CALLBACK_URL (the service with the serviceRequestUpdate endpoint, which is this microservice ia-case-payments-api)

### CODE REVIEW

File: charts/ia-case-payments-api/values.yaml [hunk 1]
```diff
@@ -8,6 +8,7 @@ java:
     S2S_URL: "http://rpe-service-auth-provider-{{ .Values.global.environment }}.service.core-compute-{{ .Values.global.environment }}.internal"
     FEES_REGISTER_API_URL: "http://fees-register-api-{{ .Values.global.environment }}.service.core-compute-{{ .Values.global.environment }}.internal"
     PAYMENT_API_URL: "http://payment-api-{{ .Values.global.environment }}.service.core-compute-{{ .Values.global.environment }}.internal"
+    PAY_CALLBACK_URL: "ia-case-payments-api-{{ .Values.global.environment }}.service.core-compute-{{ .Values.global.environment }}.internal"
```
Comment (sajjadh-v1) [inline charts/ia-case-payments-api/values.yaml:11]: Will this work without http:// ?

File: src/main/java/uk/gov/hmcts/reform/iacasepaymentsapi/domain/handlers/presubmit/PaymentAppealPreparer.java [hunk 2]
```diff
@@ -133,9 +143,23 @@ public PreSubmitCallbackResponse<AsylumCase> handle(
...
+            }
+        }
         asylumCase.write(PAYMENT_STATUS, PAYMENT_PENDING);
 
         return new PreSubmitCallbackResponse<>(asylumCase);
     }
 
+    private boolean isLegalRepJourney(AsylumCase asylumCase) {
+        return asylumCase.read(JOURNEY_TYPE, JourneyType.class)
+            .map(journey -> journey == JourneyType.REP)
```
Comment (sajjadh-v1) [inline src/main/java/uk/gov/hmcts/reform/iacasepaymentsapi/domain/handlers/presubmit/PaymentAppealPreparer.java:162]: Bit confused, so JourneyType.REP is Legal Rep, or not. If this is legal rep then perhaps orElse will be false?

File: src/main/java/uk/gov/hmcts/reform/iacasepaymentsapi/infrastructure/service/ServiceRequestService.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+@Slf4j
+public class ServiceRequestService {
+
+    private static final String PAYMENT_ACTION = "payment";
+    private static final int VOLUME_1 = 1;
+
+    private final SystemTokenGenerator systemTokenGenerator;
+    private final AuthTokenGenerator serviceAuthorization;
+    private final ServiceRequestApi serviceRequestApi;
+    String callBackUrl;
```
Comment (sajjadh-v1) [inline src/main/java/uk/gov/hmcts/reform/iacasepaymentsapi/infrastructure/service/ServiceRequestService.java:29]: Make private?

**Review by sajjadh-v1 — APPROVED** [review]
Looks good apart from a few comments made

File: src/main/java/uk/gov/hmcts/reform/iacasepaymentsapi/infrastructure/service/ServiceRequestService.java (hunk 3 above)
Comment (Marcellofabbri) [reply src/main/java/uk/gov/hmcts/reform/iacasepaymentsapi/infrastructure/service/ServiceRequestService.java]: I changed it to private.

File: src/main/java/uk/gov/hmcts/reform/iacasepaymentsapi/domain/handlers/presubmit/PaymentAppealPreparer.java (hunk 2 above)
Comment (Marcellofabbri) [reply src/main/java/uk/gov/hmcts/reform/iacasepaymentsapi/domain/handlers/presubmit/PaymentAppealPreparer.java:153]: This is the same logic that used to be in the `canHandle` methods, I just extrapolated it into a method, so I could use it inside the `isWaysToPay` method. I think it defaults to true.

But it made me realize this should be called when the appeal type is EA, HU or PA, so I added this check too.

File: charts/ia-case-payments-api/values.yaml (hunk 1 above)
Comment (Marcellofabbri) [reply charts/ia-case-payments-api/values.yaml]: Probably better to stick http:// in front of it.
I've added it

**Pull request merged**